File: domain/src/main/java/by/rekuts/travelagency/domain/CurrentUser.java
package by.rekuts.travelagency.domain;

import org.springframework.security.core.authority.AuthorityUtils;

import java.util.Objects;

public class CurrentUser extends org.springframework.security.core.userdetails.User {

    private User user;

    public CurrentUser(User user) {
        super(user.getLogin(), user.getPassword(), AuthorityUtils.createAuthorityList(user.getRole().toString()));
        this.user = user;
    }

    public User getUser() {
        return user;
    }

    public int getId() {
        return user.getId();
    }

    public User.UserRole getRole() {
        return user.getRole();
    }

    @Override
    public String toString() {
        return "CurrentUser{" +
                "user=" + user +
                "} " + super.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        CurrentUser that = (CurrentUser) o;
        return Objects.equals(user, that.user);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), user);
    }
}

File: domain/src/main/scala/by/rekuts/travelagency/domain/CurrentUser.scala
package by.rekuts.travelagency.domain

import by.rekuts.travelagency.domain.UserRole.UserRole
import org.springframework.security.core.authority.AuthorityUtils

class CurrentUser(var user: User) extends org.springframework.security.core.userdetails.User(user.login, user.password, AuthorityUtils.createAuthorityList(user.role.toString)) {
  def getUser: User = user

  def getId: Int = user.id

  def getRole: UserRole = user.role

  override def toString: String = "CurrentUser{" + "user=" + user + "} " + super.toString

  override def equals(o: Any): Boolean = {
    o match {
      case u: CurrentUser => {
        user.equals(u.getUser)
        user.id.equals(u.getId)
        user.role.equals(u.getRole)
      }
      case _ => false
    }
  }

  override def hashCode(): Int = super.hashCode()
}
